tests/utils/test_process_utils.py

```diff
@@ -54,10 +46,23 @@ def test_cache_return_value_per_process():
 
     assert len({path1, path3, f2_path1, f2_path2}) == 4
 
-    queue = Queue()
-    child_proc = Process(
-        target=_test_cache_return_value_per_process_child_proc_target, args=(path1, path3, queue)
-    )
-    child_proc.start()
-    child_proc.join()
-    assert queue.get(), "Testing inside child process failed."
+    if os.name != "nt":
+        # Test child process invalidates the cache.
+        # We don't create child process by `multiprocessing.Process` because
+        # `multiprocessing.Process` creates child process by pickling the target function
+        # and start a new process to run the pickled function. But the global variable
+        # `_per_process_value_cache_map` dict content is not pickled, this make child process
+        # automatically clear the `_per_process_value_cache_map` dict content.
```

Comment:
`child process automatically clear the `_per_process_value_cache_map` dict content.` sounds a bit strange to me. I think I'd say `_per_process_value_cache_map is empty in the child process.`

Reply:
Or child and parent don't share the same global variable.